### PR TITLE
New Jinja Functions & Better Docs

### PR DESCRIPTION
**Pull request by jpjsonic**
In this PR- 

- [x] Add new datetime jinja functions 
- [x] Add documentation for JInja Function 
- [x] Add docs for Jinja Looping 
- [x] Docs for jinja variables 
- [ ] Docs for Visivo Run Time Variables

### CODE REVIEW

File: mkdocs/reference/functions/index.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Functions
+## What Can Functions Do For You?
+Functions enable you to connect your Visivo configurations, create re-usable object, inject context and produce different outputs in different environments.
```
Comment (tgsoverly) [inline mkdocs/reference/functions/index.md:3]: What do yo mean "connect your Visivo configurations"?

File: visivo/templates/render_yaml.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
+import time
+import datetime
+from dateutil import parser
+import jinja2
+import os
+
+
+def env_var(key):
+    return os.getenv(key, "NOT-SET").replace("\\", "\\\\").replace('"', '\\"')
```
Comment (tgsoverly) [inline visivo/templates/render_yaml.py:9]: Where was this?  I think it should be `${key}-ENVIRONMENT-VARIABLE-NOT-SET` if possible.

File: visivo/utils.py [hunk 3]
```diff
@@ -93,14 +93,10 @@ def set_location_recursive_items(dictionary, file):
 
 
 def load_yaml_file(file):
-    def env_var(key):
```
Comment (tgsoverly) [inline visivo/utils.py:96]: Oh it was here... :-)

**Review by tgsoverly — APPROVED** [review]

File: visivo/templates/render_yaml.py (hunk 2 above)
Comment (jpjsonic) [reply visivo/templates/render_yaml.py:10]: I moved this function over as it was from utils.py. I can definitely update it to  say that instead.

File: mkdocs/reference/functions/index.md (hunk 1 above)
Comment (jpjsonic) [reply mkdocs/reference/functions/index.md:3]: I was talking about how ref connects objects together in the dag

**Pull request merged**